src/content/learn/manipulating-the-dom-with-refs.md

```diff
@@ -343,9 +343,11 @@ Read more about [how this helps find bugs](/reference/react/StrictMode#fixing-bu
 
 ## Accessing another component's DOM nodes {/*accessing-another-components-dom-nodes*/}
 
-When you put a ref on a built-in component that outputs a browser element like `<input />`, React will set that ref's `current` property to the corresponding DOM node (such as the actual `<input />` in the browser).
+<Pitfall>
+Refs are an escape hatch that should be used sparingly. Manually manipulating _another_ component's DOM nodes makes your code even more fragile.
+</Pitfall>
 
-However, if you try to put a ref on **your own** component, like `<MyInput />`, by default you will get `null`. Here is an example demonstrating it. Notice how clicking the button **does not** focus the input:
+When you put a ref on a built-in component that outputs a browser element like `<input />`, React will set that ref's `current` property to the corresponding DOM node (such as the actual `<input />` in the browser).
```

Comment:
This is confusing. You need to walk through these stages:

- Passing a ref to `<input>`
- Creating your own component — it doesn't work anymore
- Fixing it by spreading props

Spreading props isn't a default thing people would reach for.

Reply:
Actually it might be better to remove spreading from examples below and just do `ref={ref}` so that it's obvious it works like any other prop